Program.cs

```diff
@@ -0,0 +1,348 @@
+            sortedList = new Sort(alarms);
+            sortedList.sortList();
+        }
+    }
+    public class Event
+    {
+        private Alarm alarms = new Alarm();
+        private Sort sortedList;
+        private Field calname = new Field("X-WR-CALNAME", "", 1, 6);
+        private Field beginEvent = new Field("BEGIN", "VEVENT", -1, 7);
+        private Field dtstart = new Field("DTSTART", "", 2, 8);
+        private Field dtend = new Field("DTEND", "", 3, 9);
+        private Field dtstamp = new Field("DTSTAMP", "", 4, 10);
+        private Field uid = new Field("UID", "", 5, 11);
+        private Field description = new Field("DESCRIPTION", "", 6, 12);
+        private Field location = new Field("lOCATION", "", 7, 13);
+        private Field summary = new Field("SUMMARY", "", 8, 14);
+        private Field endEvent = new Field("END", "VEVENT", -1, 20);
+        public List<Field> events = new List<Field>();
+        public Event()
+        {
+            events = new List<Field>(new Field[] { calname, beginEvent, dtstart, dtend, dtstamp, uid, description, location, summary, endEvent });
+            sortedList = new Sort(events);
+            events.AddRange(alarms.alarms);
+            sortedList.sortList();
+        }
+        public Event(string[] values) {
+             this.events = new List<Field>(new Field[] { calname, beginEvent, dtstart, dtend, dtstamp, uid, description, location, summary, endEvent });
+             this.sortedList = new Sort(events);
+             this.events.AddRange(alarms.alarms);
+             for (int j = 0; j < values.Length; j++)  {
+                 if (this.events.Exists(x => x.getInputOrder() == j + 1)) {
+                     this.events.Find(x => x.getInputOrder().Equals(j + 1)).setValue(values[j]);
+                 }
+             }
+             this.sortedList.sortList();
+         }
+    }
+    public class TimeAndDate
+    {
+        private string icalDate;
+        private string dateString;
+        public void setDate(string date) {
+            this.icalDate = date;
+        }
+        public TimeAndDate (string date) {
+            this.icalDate = date;
+            dateString = deleteSymbolsInIcalDate(icalDate);
+            dateAndTime = format(dateString);
+        }
+        public string deleteSymbolsInIcalDate(string icalDate)
+        {
+            string dateString = icalDate;
+            for (int i = 0; i < dateString.Length; i++)
+            {
+                if (dateString[i] == 'T' || dateString[i] == 'Z')
+                {
+                    dateString = dateString.Remove(i,1);
+                }
+            }
+            return dateString;
+        }
+        public DateTime getDateAndTime() {
+            return dateAndTime;
+        }
+        private DateTime dateAndTime;
+        public DateTime format(string dateString) {
+            if(dateString.Length >0 && dateString[0] >= '0' && dateString[0] <= '9' && dateString != "2.0")
```

Comment:
We check the string being suitable to use ParseExact. It means the string has to start with decimal number, but to prevent bugs we should exclude value "2.0" because it`s a calendar version, not a date.